core/Services/Calendar/EventStore.vala

```diff
@@ -79,6 +87,29 @@ public class Calendar.EventStore : Object {
         this.month_start = Calendar.Util.datetime_get_start_of_month (get_page ());
         compute_ranges ();
 
+#if E_CAL_2_0
+        this.system_timezone = ECal.util_get_system_timezone ();
+#else
+        this.system_timezone = ECal.Util.get_system_timezone ();
+#endif
+        // Listen for system timezone changes
+        try {
+            SettingsManager settings_manager = Bus.get_proxy_sync (BusType.SYSTEM,
+                "org.freedesktop.timedate1", "/org/freedesktop/timedate1");
+            ((DBusProxy)settings_manager).g_properties_changed.connect ((changed, invalid) => {
+                var timezone = changed.lookup_value ("Timezone", GLib.VariantType.STRING);
+                if (timezone != null) {
+#if E_CAL_2_0
+                    this.system_timezone = ECal.util_get_system_timezone ();
+#else
+                    this.system_timezone = ECal.Util.get_system_timezone ();
+#endif
```

Comment:
Would it be safer to take a copy of the timezone here and when it changes?

Reply:
Definitely agreed. Fixed.